pkg/apis/recognizer/component_recognizer.go

```diff
@@ -182,6 +182,17 @@ func isAnyComponentInPath(path string, components []model.Component) bool {
 	return false
 }
 
+// isAnyComponentInDirectPath checks if a component is present in the exact path.
+// Search starts from path and will return true if a component is found.
+func isAnyComponentInDirectPath(path string, components []model.Component) bool {
+	for _, component := range components {
+		if strings.EqualFold(path, component.Path) {
```

Comment:
I would say here we could use
```golang
if strings.Contains(path, component.Path) {
```
This way we check if the container component is inside another component.